core/schema/coremod.go

```diff
@@ -142,6 +142,9 @@ func (m *CoreMod) TypeDefs(ctx context.Context) ([]*core.TypeDef, error) {
 					isIdable = true
 					continue
 				}
+				if introspectionField.Name == "sync" {
+					continue
+				}
```

Comment:
I was thinking about this - but I don't think that the sync method should appear in the `TypeDef`s. `ID` doesn't, and I think with this change, `Sync` becomes more standardized, and so is more like `ID`.

Removing it from here means the CLI doesn't see `sync` as a field property... which kind of actually makes sense to me - but this would break users who are calling `sync` on the CLI. Since we now don't default to `sync`ing... could this be an issue?

@helderco curious about your thoughts here. I'm almost thinking that if we want to do explicit syncing in the CLI, we should "raise" the syntax up, and have a special flag to do so (like `-o` today), instead of requiring `sync` to be somewhere around. This would also then allow us to implement entirely custom behavior around https://github.com/dagger/dagger/issues/7836 (and maybe do some other interesting things as well).

Reply:
I believe this is relevant:
- https://github.com/dagger/dagger/issues/7836

Reply:
As discussed in discord, we want to keep `Sync` in the typedefs for now. For consistency, we should also add it in the typedefs for user modules as well.

However, we could also potentially have `Syncable` as a property of `ObjectTypeDef`, this might also make sense.

Reply:
Argh, so I came back to this, and it's *so* weird to have a type def for user objects for this when it's not been explicitly declared by the sdk - so weird, the codebase is not structured at all to allow it, and the thought of changing it to allow it doesn't appeal really. I think maybe `sync` needs to be supported in `dagql` directly.

I'm also really reconsidering whether it should evaluate all items on the object. That also doesn't feel right to me, and additionally so because we can't do it for private fields.